src/state/cached_state.rs

```diff
@@ -102,9 +101,9 @@ impl<T: StateReader, C: ContractClassCache> CachedState<T, C> {
     }
 
     pub fn drain_private_contract_class_cache(
-        &self,
+        &mut self,
     ) -> impl Iterator<Item = (ClassHash, CompiledClass)> {
-        self.contract_class_cache_private.take().into_iter()
+        std::mem::take(&mut self.contract_class_cache_private).into_iter()
```

Comment:
```suggestion
        core::mem::take(&mut self.contract_class_cache_private).into_iter()
```
We may want to build no_std at some point, and this is free to do.